training_pipeline/models/model.py

```diff
@@ -62,82 +71,32 @@ def serving_fn(string_input):
     return serving_fn
 
 
-"""
-    _parse_tfr function is defined to parse items from TFRecord
-    files within tf.data pipeline, and _preprocess function will
-    be attatched to the tf.data pipeline to resize input images 
-    before the model
-"""
-
-
-def _parse_tfr(proto):
-    feature_description = {
-        "image": tf.io.FixedLenFeature([], tf.string),
-        "label": tf.io.FixedLenFeature([], tf.string),
-    }
-    rec = tf.io.parse_single_example(proto, feature_description)
-
-    image = tf.io.parse_tensor(rec["image"], tf.float32)
-    label = tf.io.parse_tensor(rec["label"], tf.float32)
-
-    return {"pixel_values": image, "labels": label}
-
-
-def _preprocess(example_batch):
-    images = example_batch["pixel_values"]
-    labels = tf.expand_dims(
-        example_batch["labels"], -1
-    )  # Adds extra dimension, otherwise tf.image.resize won't work.
-
-    images = tf.reshape(images, (-1, _RAW_IMG_SIZE, _RAW_IMG_SIZE, 3))
-    labels = tf.reshape(labels, (-1, _RAW_IMG_SIZE, _RAW_IMG_SIZE, 1))
-
-    images = tf.image.resize(images, (_INPUT_IMG_SIZE, _INPUT_IMG_SIZE))
-    labels = tf.image.resize(labels, (_INPUT_IMG_SIZE, _INPUT_IMG_SIZE))
-
-    labels = tf.squeeze(labels, -1)
-
-    return images, labels
-
-
 """
     _input_fn reads TFRecord files passed from the upstream 
-    TFX components such as ImportExampleGen. It calls _parse_tfr 
-    to parse each entry of the TFRecord files, then _preprocess
-    function is attached to resize the input image
+    TFX component, Transform. Assume the dataset is already 
+    transformed appropriately. 
 """
 
 
 def _input_fn(
     file_pattern: List[str],
-    batch_size: int = 32,
+    data_accessor: DataAccessor,
+    tf_transform_output: tft.TFTransformOutput,
     is_train: bool = False,
+    batch_size: int = 200,
 ) -> tf.data.Dataset:
-    INFO(f"Reading data from: {file_pattern}")
-
-    dataset = tf.data.TFRecordDataset(
-        tf.io.gfile.glob(file_pattern[0] + ".gz"),
-        num_parallel_reads=tf.data.AUTOTUNE,
-        compression_type="GZIP",
-    ).map(_parse_tfr, num_parallel_calls=tf.data.AUTOTUNE)
-
-    if is_train:
-        dataset = dataset.shuffle(batch_size * 2)
+    dataset = data_accessor.tf_dataset_factory(
+        file_pattern,
+        dataset_options.TensorFlowDatasetOptions(
+            batch_size=batch_size, label_key=_transformed_name(_LABEL_KEY)
+        ),
+        tf_transform_output.transformed_metadata.schema,
```

Comment:
Let's add informative comments here:

* How `Transform` is being used.
* How it's inferring the TFRecord schema

Anything that would help the reader read the code better.